samza-core/src/main/java/org/apache/samza/coordinator/Lock.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.samza.coordinator;
+
+public interface Lock {
+
+  /**
+   * Acquires the lock
+   */
+  boolean lock();
+
+  /**
+   * Releases the lock
+   */
+  void unlock();
+
+  boolean hasLock();
+
+  void setLockListener(LockListener listener);
```

Comment:
I personally think that lock should be synchronous utility. It makes programming much easier.
lock = getLock();

lock.lock()
.....
lock.release/unlock()